xonsh/tools.py

```diff
@@ -205,6 +216,28 @@ def __eq__(self, other):
         return all(map(operator.eq, self, other))
 
 
+class EnvPathDir(EnvPath):
```

Comment:
So that was my original implementation. But then I found out that `BASH_COMPLETIONS` uses the same class to validate/convert the paths in https://github.com/xonsh/xonsh/blob/master/xonsh/environ.py#L123. So if the file `/usr/local/etc/bash_completion` was one of the paths in `BASH_COMPLETIONS`, it would fail the directory check.